test/test_euler.py

```diff
@@ -0,0 +1,874 @@
+            vel[0] = 1.0
+            vel[1] = 1.0
+            initializer = Vortex2D(center=orig, velocity=vel)
+            #    initializer = Lump(center=orig,velocity=vel)
+            boundaries = {BTAG_ALL: PrescribedBoundary(initializer)}
+            eos = IdealSingleGas()
+
+            fields = initializer(0, nodes)
+
+            cfl = 1.0
+            sdt = get_inviscid_timestep(discr, fields, c=cfl, eos=eos)
+            constantcfl = False
+            dt = 0.001
+            nstep_status = 10
+
+            initname = initializer.__class__.__name__
+            eosname = eos.__class__.__name__
+            message = (
+                f"Num {dim}d elements: {mesh.nelements}\n"
+                f"Timestep:        {dt}\n"
+                f"Final time:      {t_final}\n"
+                f"Status freq:     {nstep_status}\n"
+                f"Initialization:  {initname}\n"
+                f"EOS:             {eosname}"
+            )
+            logging.info(message)
+
+            def write_soln():
+                expected_result = initializer(t, nodes)
+                result_resid = fields - expected_result
+                maxerr = [
+                    np.max(np.abs(result_resid[i].get()))
+                    for i in range(dim + 2)
+                ]
+                dv = eos(fields)
+                mindv = [np.min(dvfld.get()) for dvfld in dv]
+                maxdv = [np.max(dvfld.get()) for dvfld in dv]
+
+                statusmsg = (
+                    f"Status: Step({istep}) Time({t})\n"
+                    f"------   P({mindv[0]},{maxdv[0]})\n"
+                    f"------   T({mindv[1]},{maxdv[1]})\n"
+                    f"------   dt,cfl = ({dt},{cfl})\n"
+                    f"------   Err({maxerr})"
+                )
+                print(statusmsg)
+                return np.max(maxerr)
+
+            def rhs(t, w):
+                return inviscid_operator(
+                    discr, w=w, t=t, boundaries=boundaries, eos=eos
+                )
+
+            while t < t_final:
+
+                if constantcfl is True:
+                    dt = sdt
+                else:
+                    cfl = dt / sdt
+
+                if istep % nstep_status == 0:
+                    maxerr = write_soln()
+
+                fields = rk4_step(fields, t, dt, rhs)
+                t += dt
+                istep += 1
+
+                sdt = get_inviscid_timestep(
+                    discr, fields, c=cfl, eos=eos
+                )
+
+            logging.info("Writing final dump.")
+            maxerr = write_soln()
+            eoc_rec.add_data_point(1.0 / nel_1d, maxerr)
+
+        message = (
+            f"Error for (dim,order) = ({dim},{order}):\n"
+            f"{eoc_rec}"
+        )
+        logging.info(message)
+        assert (
+            eoc_rec.order_estimate() >= order - 0.5
+            or eoc_rec.max_error() < 1e-11
+        )
+
+
+# vim: foldmethod=marker
```

Comment:
This needs to be at the end of the file to make sense. (see next meeting agenda)